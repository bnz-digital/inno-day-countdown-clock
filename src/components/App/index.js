import React, { Component, Fragment } from 'react'

import Breakdown from '../Breakdown'
import Message from '../Message'
import { getData } from '../../utils'

export default class App extends Component {
  constructor (props) {
    super(props)

    const t = props.t || Date.now()

    this.state = { t }
  }

  componentDidMount () {
    this.interval = setInterval(() => this.setState({ t: Date.now() }), 100)
  }

  componentWillUnmount () {
    clearInterval(this.interval)
  }

  render () {
    const { days, hrs, mins, secs, message } = getData(this.state.t)

    return (
      <Fragment>
        <Message message={message} />
        <Breakdown days={days} hrs={hrs} mins={mins} secs={secs} />
      </Fragment>
    )
  }
}
