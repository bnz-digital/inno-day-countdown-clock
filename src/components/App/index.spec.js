import React from 'react'
import { shallow, mount } from 'enzyme'
import addDays from 'date-fns/fp/addDays'
import addHours from 'date-fns/fp/addHours'
import addMinutes from 'date-fns/fp/addMinutes'
import toDate from 'date-fns/fp/toDate'

import App from '.'
import {
  PASTRIES,
  GO_TO_BEEHIVE,
  SHOW_AND_TELL,
  INNOVATION_DAY_BEGINS,
  ID_LUNCH,
  ID_LUNCH_ENDS,
  ID_DINNER,
  ID_DINNER_ENDS,
  ID_HOT_CHOCOLATE,
  ID_HOT_CHOCOLATE_ENDS,
  ID_BREAKFAST,
  ID_BREAKFAST_ENDS,
  CODE_CUTOFF,
  LUNCH_WITH_TEAM,
  TRADE_FAIR,
  JUDGING,
  PRIZES,
  DRINKS,
  FINI
} from '../../constants'

jest.useFakeTimers()

const DAYS_BEFORE_TEST = addDays(-5)(toDate(PASTRIES))
const HOURS_BEFORE_TEST = addHours(-5)(toDate(PASTRIES))
const PASTRIES_TEST = addMinutes(5)(toDate(PASTRIES))
const GO_TO_BEEHIVE_TEST = addMinutes(5)(toDate(GO_TO_BEEHIVE))
const SHOW_AND_TELL_TEST = addMinutes(5)(toDate(SHOW_AND_TELL))
const INNOVATION_DAY_BEGINS_TEST = addMinutes(5)(toDate(INNOVATION_DAY_BEGINS))
const ID_LUNCH_TEST = addMinutes(5)(toDate(ID_LUNCH))
const ID_LUNCH_ENDS_TEST = addMinutes(5)(toDate(ID_LUNCH_ENDS))
const ID_DINNER_TEST = addMinutes(5)(toDate(ID_DINNER))
const ID_DINNER_ENDS_TEST = addMinutes(5)(toDate(ID_DINNER_ENDS))
const ID_HOT_CHOCOLATE_TEST = addMinutes(5)(toDate(ID_HOT_CHOCOLATE))
const ID_HOT_CHOCOLATE_ENDS_TEST = addMinutes(5)(toDate(ID_HOT_CHOCOLATE_ENDS))
const ID_BREAKFAST_TEST = addMinutes(5)(toDate(ID_BREAKFAST))
const ID_BREAKFAST_ENDS_TEST = addMinutes(5)(toDate(ID_BREAKFAST_ENDS))
const CODE_CUTOFF_TEST = addMinutes(5)(toDate(CODE_CUTOFF))
const LUNCH_WITH_TEAM_TEST = addMinutes(5)(toDate(LUNCH_WITH_TEAM))
const TRADE_FAIR_TEST = addMinutes(5)(toDate(TRADE_FAIR))
const JUDGING_TEST = addMinutes(5)(toDate(JUDGING))
const PRIZES_TEST = addMinutes(5)(toDate(PRIZES))
const DRINKS_TEST = addMinutes(5)(toDate(DRINKS))
const FINI_TEST = addMinutes(5)(toDate(FINI))

describe('components:App', () => {
  it('renders <App/> correctly days before the event starts', () => {
    const wrapper = shallow(<App t={DAYS_BEFORE_TEST} />)

    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders <App/> correctly hours before the event starts', () => {
    const wrapper = shallow(<App t={HOURS_BEFORE_TEST} />)

    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders <App/> correctly during PASTRIES', () => {
    const wrapper = shallow(<App t={PASTRIES_TEST} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders <App/> correctly during GO_TO_BEEHIVE', () => {
    const wrapper = shallow(<App t={GO_TO_BEEHIVE_TEST} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders <App/> correctly during SHOW_AND_TELL', () => {
    const wrapper = shallow(<App t={SHOW_AND_TELL_TEST} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders <App/> correctly right after INNOVATION_DAY_BEGINS', () => {
    const wrapper = shallow(<App t={INNOVATION_DAY_BEGINS_TEST} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders <App/> correctly during ID_LUNCH', () => {
    const wrapper = shallow(<App t={ID_LUNCH_TEST} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders <App/> correctly right after ID_LUNCH_ENDS', () => {
    const wrapper = shallow(<App t={ID_LUNCH_ENDS_TEST} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders <App/> correctly during ID_DINNER', () => {
    const wrapper = shallow(<App t={ID_DINNER_TEST} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders <App/> correctly right after ID_DINNER_ENDS', () => {
    const wrapper = shallow(<App t={ID_DINNER_ENDS_TEST} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders <App/> correctly during ID_HOT_CHOCOLATE', () => {
    const wrapper = shallow(<App t={ID_HOT_CHOCOLATE_TEST} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders <App/> correctly right after ID_HOT_CHOCOLATE_ENDS', () => {
    const wrapper = shallow(<App t={ID_HOT_CHOCOLATE_ENDS_TEST} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders <App/> correctly during ID_BREAKFAST', () => {
    const wrapper = shallow(<App t={ID_BREAKFAST_TEST} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders <App/> correctly right after ID_BREAKFAST_ENDS', () => {
    const wrapper = shallow(<App t={ID_BREAKFAST_ENDS_TEST} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders <App/> correctly right after CODE_CUTOFF', () => {
    const wrapper = shallow(<App t={CODE_CUTOFF_TEST} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders <App/> correctly during LUNCH_WITH_TEAM', () => {
    const wrapper = shallow(<App t={LUNCH_WITH_TEAM_TEST} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders <App/> correctly during TRADE_FAIR', () => {
    const wrapper = shallow(<App t={TRADE_FAIR_TEST} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders <App/> correctly during JUDGING', () => {
    const wrapper = shallow(<App t={JUDGING_TEST} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders <App/> correctly during PRIZES', () => {
    const wrapper = shallow(<App t={PRIZES_TEST} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders <App/> correctly during DRINKS', () => {
    const wrapper = shallow(<App t={DRINKS_TEST} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders <App/> correctly after FINI', () => {
    const wrapper = shallow(<App t={FINI_TEST} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('runs clearInterval on unmount', () => {
    const wrapper = mount(<App />)
    jest.runOnlyPendingTimers()
    expect(setInterval).toHaveBeenCalled()
    wrapper.unmount()
    expect(clearInterval).toHaveBeenCalledTimes(1)
  })
})
