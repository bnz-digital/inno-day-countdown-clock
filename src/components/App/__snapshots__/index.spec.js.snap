// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:App renders <App/> correctly after FINI 1`] = `
<React.Fragment>
  <Message
    message={
      <Unknown>
        <StyledHighlight>
          Show's over, folks!
        </StyledHighlight>
        <StyledLocation>
          Gumbles has left the building.
        </StyledLocation>
      </Unknown>
    }
  />
  <Breakdown
    hrs="00"
    mins="00"
    secs="00"
  />
</React.Fragment>
`;

exports[`components:App renders <App/> correctly days before the event starts 1`] = `
<React.Fragment>
  <Message
    message={
      <Unknown>
        <Unknown>
          Less than
           
          <StyledHighlight>
            6 days
          </StyledHighlight>
           
          to
           
          <StyledHighlight>
            Innovation Day!
          </StyledHighlight>
        </Unknown>
        <StyledLocation>
          Pastries at 7:45 AM on Coop levels 1, 9, and 14
        </StyledLocation>
        <p>
          Are you ready?
        </p>
      </Unknown>
    }
  />
  <Breakdown
    days={5}
    hrs="00"
    mins="00"
    secs="00"
  />
</React.Fragment>
`;

exports[`components:App renders <App/> correctly during DRINKS 1`] = `
<React.Fragment>
  <Message
    message={
      <Unknown>
        <p>
          <StyledHighlight>
            The Wrap Party
          </StyledHighlight>
           is happening
           
          <StyledHighlight>
            NOW!
          </StyledHighlight>
        </p>
        <StyledLocation>
          4:30-8 PM, Te Aro Room, Mac's on the Waterfront
        </StyledLocation>
      </Unknown>
    }
  />
  <Breakdown
    days={0}
    hrs="03"
    mins="25"
    secs="00"
  />
</React.Fragment>
`;

exports[`components:App renders <App/> correctly during GO_TO_BEEHIVE 1`] = `
<React.Fragment>
  <Message
    message={
      <Unknown>
        <Unknown>
          Less than
           
          <StyledHighlight>
            26 minutes
          </StyledHighlight>
           
          to
           
          <StyledHighlight>
            Show & Tell
          </StyledHighlight>
        </Unknown>
        <StyledLocation>
          9 AM at the Beehive Banquet Hall
        </StyledLocation>
        <Now>
          Make way to the Beehive!
        </Now>
      </Unknown>
    }
  />
  <Breakdown
    days={0}
    hrs="00"
    mins="25"
    secs="00"
  />
</React.Fragment>
`;

exports[`components:App renders <App/> correctly during ID_BREAKFAST 1`] = `
<React.Fragment>
  <Message
    message={
      <Unknown>
        <Unknown>
          Less than
           
          <StyledHighlight>
            26 minutes
          </StyledHighlight>
           
          to
           
          <StyledHighlight>
            breakfast ends
          </StyledHighlight>
        </Unknown>
        <StyledLocation>
          Come to Coop levels 1, 9, or 14
        </StyledLocation>
        <Now>
          Pastries and fruit!
        </Now>
        <StyledSubTimer>
          <StyledHighlight>
            Code Cutoff in
             
            <Breakdown
              days={0}
              hrs="03"
              mins="25"
              secs="00"
              unstyled={true}
            />
          </StyledHighlight>
        </StyledSubTimer>
      </Unknown>
    }
  />
  <Breakdown
    days={0}
    hrs="00"
    mins="25"
    secs="00"
  />
</React.Fragment>
`;

exports[`components:App renders <App/> correctly during ID_DINNER 1`] = `
<React.Fragment>
  <Message
    message={
      <Unknown>
        <Unknown>
          Less than
           
          <StyledHighlight>
            26 minutes
          </StyledHighlight>
           
          to
           
          <StyledHighlight>
            dinner's over
          </StyledHighlight>
        </Unknown>
        <StyledLocation>
          Come to the Coop House Atria Caf√©
        </StyledLocation>
        <Now>
          The Big Digital Dinner!
        </Now>
        <StyledSubTimer>
          <StyledHighlight>
            Code Cutoff in
             
            <Breakdown
              days={0}
              hrs="16"
              mins="25"
              secs="00"
              unstyled={true}
            />
          </StyledHighlight>
        </StyledSubTimer>
      </Unknown>
    }
  />
  <Breakdown
    days={0}
    hrs="00"
    mins="25"
    secs="00"
  />
</React.Fragment>
`;

exports[`components:App renders <App/> correctly during ID_HOT_CHOCOLATE 1`] = `
<React.Fragment>
  <Message
    message={
      <Unknown>
        <Unknown>
          Less than
           
          <StyledHighlight>
            8 hours
          </StyledHighlight>
           
          to
           
          <StyledHighlight>
            Breakfast
          </StyledHighlight>
        </Unknown>
        <StyledLocation>
          Pastries at 7:30 AM on Coop levels 1, 9, and 14
        </StyledLocation>
        <Now>
          Midnight hot chocolate in LG.P1.20
        </Now>
        <StyledSubTimer>
          <StyledHighlight>
            Code Cutoff in
             
            <Breakdown
              days={0}
              hrs="10"
              mins="55"
              secs="00"
              unstyled={true}
            />
          </StyledHighlight>
        </StyledSubTimer>
      </Unknown>
    }
  />
  <Breakdown
    days={0}
    hrs="00"
    mins="25"
    secs="00"
  />
</React.Fragment>
`;

exports[`components:App renders <App/> correctly during ID_LUNCH 1`] = `
<React.Fragment>
  <Message
    message={
      <Unknown>
        <Unknown>
          Less than
           
          <StyledHighlight>
            26 minutes
          </StyledHighlight>
           
          to
           
          <StyledHighlight>
            grab lunch
          </StyledHighlight>
        </Unknown>
        <StyledLocation>
          Coop levels 1, 9, and 14
        </StyledLocation>
        <Now>
          Pick up your Eat My Lunch!
        </Now>
        <StyledSubTimer>
          <StyledHighlight>
            Code Cutoff in
             
            <Breakdown
              days={0}
              hrs="22"
              mins="25"
              secs="00"
              unstyled={true}
            />
          </StyledHighlight>
        </StyledSubTimer>
      </Unknown>
    }
  />
  <Breakdown
    days={0}
    hrs="00"
    mins="25"
    secs="00"
  />
</React.Fragment>
`;

exports[`components:App renders <App/> correctly during JUDGING 1`] = `
<React.Fragment>
  <Message
    message={
      <Unknown>
        <Unknown>
          Less than
           
          <StyledHighlight>
            26 minutes
          </StyledHighlight>
           
          to
           
          <StyledHighlight>
            Prizes!
          </StyledHighlight>
        </Unknown>
        <StyledLocation>
          4 PM, Te Aro Room, Mac's on the Waterfront
        </StyledLocation>
        <Now>
          Take a break while the judges deliberate!
        </Now>
      </Unknown>
    }
  />
  <Breakdown
    days={0}
    hrs="00"
    mins="25"
    secs="00"
  />
</React.Fragment>
`;

exports[`components:App renders <App/> correctly during LUNCH_WITH_TEAM 1`] = `
<React.Fragment>
  <Message
    message={
      <Unknown>
        <Unknown>
          Less than
           
          <StyledHighlight>
            86 minutes
          </StyledHighlight>
           
          to
           
          <StyledHighlight>
            the Trade Fair!
          </StyledHighlight>
        </Unknown>
        <StyledLocation>
          1:30 PM, Te Aro Room, Mac's on the Waterfront
        </StyledLocation>
        <StyledLocation>
          NOTE: Macs Brewery is open for project set up
        </StyledLocation>
        <Now>
          <StyledHighlight>
            Take your team to lunch!
          </StyledHighlight>
        </Now>
      </Unknown>
    }
  />
  <Breakdown
    days={0}
    hrs="01"
    mins="25"
    secs="00"
  />
</React.Fragment>
`;

exports[`components:App renders <App/> correctly during PASTRIES 1`] = `
<React.Fragment>
  <Message
    message={
      <Unknown>
        <Unknown>
          Less than
           
          <StyledHighlight>
            41 minutes
          </StyledHighlight>
           
          to
           
          <StyledHighlight>
            head to the Beehive
          </StyledHighlight>
        </Unknown>
        <StyledLocation>
          At 8:30 AM make way to the Beehive Banquet Hall
        </StyledLocation>
        <Now>
          Pastries on Coop levels 1, 9, and 14
        </Now>
      </Unknown>
    }
  />
  <Breakdown
    days={0}
    hrs="00"
    mins="40"
    secs="00"
  />
</React.Fragment>
`;

exports[`components:App renders <App/> correctly during PRIZES 1`] = `
<React.Fragment>
  <Message
    message={
      <Unknown>
        <Unknown>
          Less than
           
          <StyledHighlight>
            26 minutes
          </StyledHighlight>
           
          to
           
          <StyledHighlight>
            the Wrap Party!
          </StyledHighlight>
        </Unknown>
        <StyledLocation>
          4:30 PM, Te Aro Room, Mac's on the Waterfront
        </StyledLocation>
        <Now>
          Prizes at Mac's on the Waterfront!
        </Now>
      </Unknown>
    }
  />
  <Breakdown
    days={0}
    hrs="00"
    mins="25"
    secs="00"
  />
</React.Fragment>
`;

exports[`components:App renders <App/> correctly during SHOW_AND_TELL 1`] = `
<React.Fragment>
  <Message
    message={
      <Unknown>
        <Unknown>
          Less than
           
          <StyledHighlight>
            2 hours
          </StyledHighlight>
           
          to
           
          <StyledHighlight>
            Innovation Day
          </StyledHighlight>
        </Unknown>
        <StyledLocation>
          11 AM at the Coop Building
        </StyledLocation>
        <Now>
          Show & Tell at the Beehive Banquet Hall
        </Now>
      </Unknown>
    }
  />
  <Breakdown
    days={0}
    hrs="01"
    mins="55"
    secs="00"
  />
</React.Fragment>
`;

exports[`components:App renders <App/> correctly during TRADE_FAIR 1`] = `
<React.Fragment>
  <Message
    message={
      <Unknown>
        <Unknown>
          Less than
           
          <StyledHighlight>
            2 hours
          </StyledHighlight>
           
          to
           
          <StyledHighlight>
            the Judging
          </StyledHighlight>
        </Unknown>
        <StyledLocation>
          3:30 PM, Te Aro Room, Mac's on the Waterfront
        </StyledLocation>
        <Now>
          Trade Fair at Mac's (Waterfront)
        </Now>
      </Unknown>
    }
  />
  <Breakdown
    days={0}
    hrs="01"
    mins="55"
    secs="00"
  />
</React.Fragment>
`;

exports[`components:App renders <App/> correctly hours before the event starts 1`] = `
<React.Fragment>
  <Message
    message={
      <Unknown>
        <Unknown>
          Less than
           
          <StyledHighlight>
            6 hours
          </StyledHighlight>
           
          to
           
          <StyledHighlight>
            Innovation Day!
          </StyledHighlight>
        </Unknown>
        <StyledLocation>
          Pastries at 7:45 AM on Coop levels 1, 9, and 14
        </StyledLocation>
        <p>
          Are you ready?
        </p>
      </Unknown>
    }
  />
  <Breakdown
    days={0}
    hrs="05"
    mins="00"
    secs="00"
  />
</React.Fragment>
`;

exports[`components:App renders <App/> correctly right after CODE_CUTOFF 1`] = `
<React.Fragment>
  <Message
    message={
      <Unknown>
        <Unknown>
          Less than
           
          <StyledHighlight>
            3 hours
          </StyledHighlight>
           
          to
           
          <StyledHighlight>
            the Trade Fair!
          </StyledHighlight>
        </Unknown>
        <StyledLocation>
          1:30 PM, Te Aro Room, Mac's on the Waterfront
        </StyledLocation>
        <StyledLocation>
          NOTE: Macs Brewery is open for project set up
        </StyledLocation>
        <Now>
          <StyledHighlight>
            Take a breather
          </StyledHighlight>
        </Now>
      </Unknown>
    }
  />
  <Breakdown
    days={0}
    hrs="02"
    mins="25"
    secs="00"
  />
</React.Fragment>
`;

exports[`components:App renders <App/> correctly right after ID_BREAKFAST_ENDS 1`] = `
<React.Fragment>
  <Message
    message={
      <Unknown>
        <Unknown>
          Less than
           
          <StyledHighlight>
            3 hours
          </StyledHighlight>
           
          to
           
          <StyledHighlight>
            Code Cutoff!
          </StyledHighlight>
        </Unknown>
        <p>
          Finish up!
        </p>
      </Unknown>
    }
  />
  <Breakdown
    days={0}
    hrs="02"
    mins="55"
    secs="00"
  />
</React.Fragment>
`;

exports[`components:App renders <App/> correctly right after ID_DINNER_ENDS 1`] = `
<React.Fragment>
  <Message
    message={
      <Unknown>
        <Unknown>
          Less than
           
          <StyledHighlight>
            5 hours
          </StyledHighlight>
           
          to
           
          <StyledHighlight>
            Midnight Hot Chocolate!
          </StyledHighlight>
        </Unknown>
        <StyledLocation>
          At Midnight in LG.P1.20
        </StyledLocation>
        <Now>
          Chill out time in LG.P1.20
        </Now>
        <StyledSubTimer>
          <StyledHighlight>
            Code Cutoff in
             
            <Breakdown
              days={0}
              hrs="15"
              mins="55"
              secs="00"
              unstyled={true}
            />
          </StyledHighlight>
        </StyledSubTimer>
      </Unknown>
    }
  />
  <Breakdown
    days={0}
    hrs="04"
    mins="55"
    secs="00"
  />
</React.Fragment>
`;

exports[`components:App renders <App/> correctly right after ID_HOT_CHOCOLATE_ENDS 1`] = `
<React.Fragment>
  <Message
    message={
      <Unknown>
        <Unknown>
          Less than
           
          <StyledHighlight>
            7 hours
          </StyledHighlight>
           
          to
           
          <StyledHighlight>
            Breakfast
          </StyledHighlight>
        </Unknown>
        <StyledLocation>
          Pastries at 7:30 AM on Coop levels 1, 9, and 14
        </StyledLocation>
        <Now>
          Chill out time in LG.P1.20
        </Now>
        <StyledSubTimer>
          <StyledHighlight>
            Code Cutoff in
             
            <Breakdown
              days={0}
              hrs="10"
              mins="25"
              secs="00"
              unstyled={true}
            />
          </StyledHighlight>
        </StyledSubTimer>
      </Unknown>
    }
  />
  <Breakdown
    days={0}
    hrs="06"
    mins="55"
    secs="00"
  />
</React.Fragment>
`;

exports[`components:App renders <App/> correctly right after ID_LUNCH_ENDS 1`] = `
<React.Fragment>
  <Message
    message={
      <Unknown>
        <Unknown>
          Less than
           
          <StyledHighlight>
            6 hours
          </StyledHighlight>
           
          to
           
          <StyledHighlight>
            Digital Dinner
          </StyledHighlight>
        </Unknown>
        <StyledLocation>
          At 6:30 PM in the Coop House Atria Caf√©
        </StyledLocation>
        <Now>
          Work on your Innovation Day project!
        </Now>
        <StyledSubTimer>
          <StyledHighlight>
            Code Cutoff in
             
            <Breakdown
              days={0}
              hrs="21"
              mins="55"
              secs="00"
              unstyled={true}
            />
          </StyledHighlight>
        </StyledSubTimer>
      </Unknown>
    }
  />
  <Breakdown
    days={0}
    hrs="05"
    mins="25"
    secs="00"
  />
</React.Fragment>
`;

exports[`components:App renders <App/> correctly right after INNOVATION_DAY_BEGINS 1`] = `
<React.Fragment>
  <Message
    message={
      <Unknown>
        <Unknown>
          Less than
           
          <StyledHighlight>
            86 minutes
          </StyledHighlight>
           
          to
           
          <StyledHighlight>
            Eat My Lunch!
          </StyledHighlight>
        </Unknown>
        <StyledLocation>
          12:30 PM on Coop levels 1, 9, and 14
        </StyledLocation>
        <Now>
          Work on your Innovation Day project!
        </Now>
        <StyledSubTimer>
          <StyledHighlight>
            Code Cutoff in
             
            <Breakdown
              days={0}
              hrs="23"
              mins="55"
              secs="00"
              unstyled={true}
            />
          </StyledHighlight>
        </StyledSubTimer>
      </Unknown>
    }
  />
  <Breakdown
    days={0}
    hrs="01"
    mins="25"
    secs="00"
  />
</React.Fragment>
`;
